### ISSUE TITLE
Index out of bounds on i64. 

### ISSUE THREAD

**Issue by crgant12** [bluejekyll/foundationdb-rs#119]
I'm trying to encode `-8617230260136600747` in a tuple, but it panics when encoding. Here is my example code.
```extern crate foundationdb;

use foundationdb::tuple::{Tuple,Element,Encode};

fn main() {
    println!("Encoding -8617230260136600747");
    let mut vec = Vec::new();
    let tup = Tuple::from(vec![Element::I64(-8617230260136600747)]).encode_to(&mut vec);
    println!("Encoded tup of cell = {:?}", tup);
}
```

Here is the output

```
Encoding -8617230260136600747
thread 'main' panicked at 'index out of bounds: the len is 8 but the index is 8', /Users/chris/.cargo/registry/src/github.com-1ecc6299db9ec823/foundationdb-0.3.0/src/tuple/element.rs:417:48
```

I usually use Go when interacting with foundation, so I looked at what could be different and in go the `SIZE_LIMIT` goes up to `1<<(8*8) - 1` coming from here https://github.com/apple/foundationdb/blob/master/bindings/go/src/fdb/tuple/tuple.go#L125-L135 while the rust doesn't since it is only an `i64` and not a `u64`

**Comment by bluejekyll** [comment]
Thank you for the report. Is this on the master branch?

Also, there is an open request to add support for u64: #117

**Comment by crgant12** [comment]
Yes, this is on the master branch. Also I mentioned u64 for the `SIZE_LIMIT` as it is a list of i64, so adding ` (1 << (8 * 8)) - 1` would cause it to overflow.

**Comment by bluejekyll** [comment]
If you'd be interested in contributing a patch, that would be greatly appreciated. I'm not going to be able to work on this for a little bit. @yjh0502 maybe you have some interest?

**Comment by Speedy37** [comment]
This is fixed in both the 0.3 and the master (async / 0.4) branches
